jina/peapods/runtimes/asyncio/rest/app.py

```diff
@@ -18,6 +18,21 @@
 from .....types.request import Request
 
 
+# TODO instead of this function make sure that only one 'oneof field' is used.
```

Comment:
Pydantic's `oneof` implementation is based on python `Union` and represents something different than the protobuf implementation. We need to have custom logic in `protobuf_to_pydantic_model` to make `oneof` pick only one field.

https://github.com/samuelcolvin/pydantic/issues/656
https://github.com/samuelcolvin/pydantic/issues/619

Reply:
Good to see that others are interested in this issue as well. 
Until they don't support this feature, we have to go with this approach I guess.